### PR TITLE
widget: Replace old-style Qt connect() calls

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Comment by uklotzde** [comment]
I will file a PR with suggested fixes for some legacy issues here.

**Review by uklotzde — COMMENTED** [review]
WBattery::update() hides the non-virtual public slot QWidget::update(). That was wrong.

I have pushed the suggested changes directly onto this branch, please have a look.

**Review by uklotzde — APPROVED** [review]

File: src/widget/wknob.cpp [hunk 1]
```diff
@@ -26,9 +26,10 @@ WKnob::WKnob(QWidget* pParent)
         : WDisplay(pParent),
           m_renderTimer(mixxx::Duration::fromMillis(20),
                         mixxx::Duration::fromSeconds(1)) {
-    connect(&m_renderTimer, &WidgetRenderTimer::update, this, [this]() {
-        update();
-    });
+    connect(&m_renderTimer,
+            &WidgetRenderTimer::update,
+            this,
+            QOverload<>::of(&QWidget::update));
```
Comment (Holzhaus) [inline src/widget/wknob.cpp:32]: Ah, that's how it works... :bulb: I tried `QOverload<void>` but that failed.

**Review by Holzhaus — COMMENTED** [review]
LGTM, thank you!

**Pull request merged**